R/object_to_R_code.R

```diff
@@ -0,0 +1,36 @@
+#' Transform a vector into a string that evaluates to the same vector
+#'
+#' Collapses a vector into a comma-separated list with strings quoted
+#' (and special characters escaped). A general purpose helper function
+#' for writing new model code.
+#'
+#' @param vec a vector
+#' @param return_null logical indicating if `vec` is NULL whether to "NULL"
+#'   (alternative is to return an empty list).
+#' @return character string of length 1
+
+vector_to_R_code <- function(vec, return_null = TRUE) {
+  if (return_null && is.null(vec)) return("NULL")
```

Comment:
Does the model code actually behave differently if we use `c()` vs. `NULL`, since both evaluate to `NULL`?

Reply:
hmmm, maybe not 🤔 I can't get it to fail in a few examples of models I've tried. Do we like `NULL` or `c()` better?

Reply:
If we do `c()` we can remove this line and just have the `paste0()` call, OTOH `NULL` seems a little more explicit/readable. I don't feel too strongly one way or the other.